PROTOCOL.md

```diff
@@ -1754,13 +1754,16 @@ Type | Serialization Format
 string | No translation required
 numeric types | The string representation of the number
 date | Encoded as `{year}-{month}-{day}`. For example, `1970-01-01`
-timestamp | Encoded as `{year}-{month}-{day} {hour}:{minute}:{second}` or `{year}-{month}-{day} {hour}:{minute}:{second}.{microsecond}` For example: `1970-01-01 00:00:00`, or `1970-01-01 00:00:00.123456`
+timestamp | Encoded as `{year}-{month}-{day} {hour}:{minute}:{second}` or `{year}-{month}-{day} {hour}:{minute}:{second}.{microsecond}`. Each of these options may also be suffixed with UTC for UTC adjusted timezones. For example: `1970-01-01 00:00:00`, or `1970-01-01 00:00:00.123456` or `2024-06-14 04:00:00UTC` or `2024-06-14 04:00:00.123456UTC`
```

Comment:
do we also want to put the Z here ?  eg 2024-06-14 04:00:00Z ?
I do not have opinion..

Reply:
I think we can allow both. Using Z is actually slightly better because it's shorter. :)

Reply:
I was thinking about this a bit more, we probably want the protocol to just be pretty strict here since imo there's not much value in enabling multiple ways to store the partition value. Since we are just now defining UTC adjusted timestamps as allowed, we don't have to worry about compatibility and being strict ensures that everyone is speaking the same language here.

I agree that "Z" is better, not only because it's shorter but also we probably want to stick to well established industry standards such as ISO8601 where there's no concept of adding the "UTC" suffix. It must be "Z". 

So I've updated the protocol proposal change so that the UTC adjusted timestamp must be stored as ISO8601 format.

This has some other implications:

ISO8601 does *not* allow space between the time and the date. There must be a "T" character. 

Again, if you compare this format to the current format which is not UTC adjusted it's a bit different but I think it's better since ISO8601 is already a standard and people's existing timestamp formatters would *just work*.

Reply:
I was thinking about this a bit more, we probably want the protocol to just be pretty strict here since imo there's not much value in enabling multiple ways to store the partition value. Since we are just now defining UTC adjusted timestamps as allowed, we don't have to worry about compatibility and being strict ensures that everyone is speaking the same language here.

I agree that "Z" is better, not only because it's shorter but also we probably want to stick to well established industry standards such as ISO8601 where there's no concept of adding the "UTC" suffix. It must be "Z". 

So I've updated the protocol proposal change so that the UTC adjusted timestamp must be stored as ISO8601 format.

This has some other implications:

ISO8601 does *not* allow space between the time and the date. There must be a "T" character. 

Again, if you compare this format to the current format which is not UTC adjusted it's a bit different but I think it's better since ISO8601 is already a standard and people's existing timestamp formatters would *just work*.